RR.App.Tests/Controllers/HRIS/EmployeeBankingControllerUnitTests.cs

```diff
@@ -151,65 +91,77 @@ public async Task GetValidStatusReturnsOkResultWithEntries()
     }
 
     [Fact]
-    public async Task GetExceptionThrownReturnsNotFoundResultWithErrorMessage()
+    public async Task UpdateValidDataReturnsOkResult()
     {
-        _employeeBankingServiceMock.Setup(x => x.Get(1)).ThrowsAsync(new Exception("Unable to retrieve employee banking entries"));
+        _employeeBankingServiceMock.Setup(x => x.Update(It.IsAny<EmployeeBankingDto>(), "test@example.com"))
+                   .ReturnsAsync(_employeeBankingDto);
 
-        var result = await _controller.Get(1);
+        var result = await _controller.Update(_employeeBankingDto);
 
-        var notFoundResult = Assert.IsType<NotFoundObjectResult>(result);
-        var actualErrorMessage = Assert.IsType<string>(notFoundResult.Value);
-        Assert.Equal("Unable to retrieve employee banking entries", actualErrorMessage);
+        var okResult = Assert.IsType<OkResult>(result);
+        Assert.Equal(200, okResult.StatusCode);
     }
 
     [Fact]
-    public async Task UpdateValidDataReturnsOkResult()
+    public async Task GetBankingDetailsReturnsOkResult()
     {
-        _employeeBankingServiceMock.Setup(x => x.Update(It.IsAny<EmployeeBankingDto>(), "test@example.com"))
-                   .ReturnsAsync(_employeeBankingDto);
+        _identity.Setup(i => i.Role).Returns("Admin");
+        _identity.SetupGet(i => i.EmployeeId).Returns(2);
+        _identity.SetupGet(i => i.Email).Returns("admin@example.com");
 
-        var result = await _controller.Update(_simpleEmployeeBankingDto);
+        _employeeBankingServiceMock.Setup(x => x.GetBanking(_employeeBankingDto.EmployeeId))
+                                   .ReturnsAsync(_employeeBankingDtoList);
 
-        var okResult = Assert.IsType<OkResult>(result);
-        Assert.Equal(200, okResult.StatusCode);
+        var result = await _controller.GetBankingDetails(_employeeBankingDto.EmployeeId);
+
+        var okResult = Assert.IsType<OkObjectResult>(result);
+        var returnValue = Assert.IsType<List<EmployeeBankingDto>>(okResult.Value);
+        Assert.Equal(_employeeBankingDtoList, returnValue);
+        _employeeBankingServiceMock.Verify(service => service.GetBanking(_employeeBankingDto.EmployeeId), Times.Once);
     }
 
     [Fact]
-    public async Task UpdateExceptionThrownReturnsNotFoundResultWithErrorMessage()
+    public async Task GetBankingUnauthorizedAccess()
     {
-        _employeeBankingServiceMock.Setup(x => x.Update(It.IsAny<EmployeeBankingDto>(), "test@example.com"))
-                   .ThrowsAsync(new Exception("Some error message"));
+        _identity.Setup(i => i.Role).Returns("Talent");
+        _identity.SetupGet(i => i.EmployeeId).Returns(5);
+        _identity.SetupGet(i => i.Email).Returns("talent@example.com");
```

Comment:
maybe setup the _identity at class level
![image](https://github.com/user-attachments/assets/d257d03e-cf83-4c1c-81be-870e3768a312)